woodwork/table_accessor.py

```diff
@@ -617,10 +657,10 @@ def _get_subset_df_with_schema(self, cols_to_include, inplace=False):
         if inplace:
             cols_to_drop = [col_name for col_name in self._dataframe.columns if col_name not in cols_to_include]
             self._dataframe.drop(cols_to_drop, axis='columns', inplace=True)
-            self.init(schema=new_schema, validate=False)
+            self.init_with_full_schema(schema=new_schema, validate=False)
             return
         new_df = self._dataframe[cols_to_include]
-        new_df.ww.init(schema=new_schema, validate=False)
+        new_df.ww.init_with_partial_schema(schema=new_schema, validate=False)
```

Comment:
I believe `new_schema` will be the full schema for `new_df` since the subset used in line 656 is the same as what was used to create `new_df`, so `init_with_full_schema` may be more clear here